### PR TITLE
Feat: Add clear formatting button to playground

### LINKED ISSUE #2190: Feature: Add a "Clear formatting" button to the Playground
Add a "Clear formatting" to the Playground which will restore all text content to an unstyled paragraph.

### PR DESCRIPTION
**Pull request by im-adithya**
Closes #2190

https://user-images.githubusercontent.com/64399555/170961978-128faa38-23e9-4cdb-8e03-63e5f64d8f11.mp4

**Note:** Is it okay to leave the hashtag as is? Because lexical changes the node from text to hashtag whenever a string starts with "#". But due to changing the background color to default, the light blue background of the hashtag is being removed.

### CODE REVIEW

**Comment by trueadm** [comment]
Could use an e2e test to confirm it works right?

**Comment by im-adithya** [comment]
@trueadm Can you please let me know about handling the hashtag node as discussed above? That would help me in writing e2e for this functionality!

File: packages/lexical-playground/src/plugins/ToolbarPlugin.tsx [hunk 1]
```diff
@@ -920,6 +922,29 @@ export default function ToolbarPlugin(): JSX.Element {
...
+          'background-color': '#fff',
+          color: '#000',
+          'font-family': 'Arial',
+          'font-size': '15px',
+        });
+        $wrapLeafNodesInElements(selection, () => $createParagraphNode());
+      }
+      activeEditor.dispatchCommand(TOGGLE_LINK_COMMAND, null);
+      activeEditor.dispatchCommand(FORMAT_ELEMENT_COMMAND, 'left');
+    });
```
Comment (thegreatercurve) [inline packages/lexical-playground/src/plugins/ToolbarPlugin.tsx:945]: Apologies, I was basing the original behaviour for this on the "Paste without format" behaviour in Google Docs and Word, but the "Clear All Formatting" behaviour is slightly different in that it doesn't necessarily remove links, hashtags, or list elements. 

It just clears the format and any text color, but NOT the size. More from [the Word docs here](https://support.microsoft.com/en-us/office/clear-all-text-formatting-c094c4da-7f09-4cea-9a8d-c166949c9c80): 
> You can easily clear all formatting (such as bold, underline, italics, color, superscript, subscript, and more) from your text and return your text to its default formatting styles.

I think therefore we can just do the below:

```suggestion
    activeEditor.update(() => {
      const selection = $getSelection();
      if ($isRangeSelection(selection)) {
        $selectAll(selection);
        selection.getNodes().forEach((node) => {
          if ($isTextNode(node)) {
            node.setFormat(0);
          }
        });
        $patchStyleText(selection, {
          'background-color': '#fff',
          color: '#000',
          'font-family': 'Arial',
        });
      }
      activeEditor.dispatchCommand(FORMAT_ELEMENT_COMMAND, 'left');
    });
```

File: packages/lexical-playground/src/plugins/ToolbarPlugin.tsx [hunk 2]
```diff
@@ -975,6 +1000,13 @@ export default function ToolbarPlugin(): JSX.Element {
 
   return (
     <div className="toolbar">
+      <button
+        onClick={clearFormatting}
+        title="Clear Formatting"
+        className="toolbar-item spaced"
+        aria-label="Clear">
```
Comment (thegreatercurve) [inline packages/lexical-playground/src/plugins/ToolbarPlugin.tsx:1007]:
```suggestion
        aria-label="Clear all text formatting">
```

File: packages/lexical-playground/src/plugins/ToolbarPlugin.tsx [hunk 3]
```diff
@@ -975,6 +1000,13 @@ export default function ToolbarPlugin(): JSX.Element {
 
   return (
     <div className="toolbar">
+      <button
+        onClick={clearFormatting}
+        title="Clear Formatting"
```
Comment (thegreatercurve) [inline packages/lexical-playground/src/plugins/ToolbarPlugin.tsx:1005]: Might be a little clearer for accessibility: 
```suggestion
        title="Clear text formatting"
```

**Review by thegreatercurve — CHANGES_REQUESTED** [review]
LGTM! Just clarified some of the behaviour in the comments above which should also answer the hashtag question

File: packages/lexical-playground/src/plugins/ToolbarPlugin.tsx [hunk 4]
```diff
@@ -975,6 +1000,13 @@ export default function ToolbarPlugin(): JSX.Element {
 
   return (
     <div className="toolbar">
+      <button
+        onClick={clearFormatting}
```
[the remark this answers is not in the document]
Comment (im-adithya) [reply packages/lexical-playground/src/plugins/ToolbarPlugin.tsx]: Thanks for the valuable inputs! Made all the necessary changes!

**Comment by im-adithya** [comment]
@thegreatercurve shouldn't we leave the nodes which already have some pre-specified text-color (quote node and link node) or background-color (hashtags, mentions) while using the `$patchStyleText` helper function? Attaching these screenshots for reference:

Before Clearing:
Screenshot from 2022-05-30 21-20-36

After Clearing:
Screenshot from 2022-05-30 21-20-33

**Comment by thegreatercurve** [comment]
> @thegreatercurve shouldn't we leave the nodes which already have some pre-specified text-color (quote node and link node) or background-color (hashtags, mentions) while using the `$patchStyleText` helper function? Attaching these screenshots for reference:
> 
> Before Clearing: Screenshot from 2022-05-30 21-20-36
> 
> After Clearing: Screenshot from 2022-05-30 21-20-33

@im-adithya Very good point. You're right, in other editors, they essentially reset the non-TextNodes to its original styles. 

In that case, I think we can still solve the issue with the existing APIs. There is a `setStyle` method on TextNode, which we can use to reset things such as font, and highlight or text colors. 

So, if you remove `$patchStyleText`, you can then do something like the below in your loop:

```js
 if ($isTextNode(node) || $isHashTahNode(node) || $isMentionNode(node)) {
    node.setFormat(0);
    node.setStyle('');
 }
```

**Comment by im-adithya** [comment]
Done! Checking for just `$textNode` worked fine as the other two were extending from it!

**Review by thegreatercurve — APPROVED** [review]
LGTM. Thanks for working on this @im-adithya.

**Pull request merged**